### ISSUE TITLE
[UX] Make the 'Formatting options' link less distracting

### ISSUE THREAD

**Issue by bradbulger** [backdrop/backdrop-issues#3516]
ORIGINAL ISSUE:
These started as comments from a client, I largely agree though. The overall point is that once the content types get to a certain level of complexity, the node forms begin to look awkward in the default Seven styles. Buttons and other elements that are there for the user to perform some kind of action aren't always easy to see and tell what they are.

For instance, on a text field with unlimited values, the "ADD ANOTHER ITEM" button is the same size, shape, and color as the field title. Inside a field group container like a fieldset or vertical tab, the field title isn't even bolded. If you have several of these fields in a form, it gets confusing visually.

Another example the client raised was the size of the FORMATTING OPTIONS fieldset label. The default `legend` font size is 120% - that's fine for fieldsets containing fields, but in a largely information-only box repeated for every formatted textarea on the form, it's a bit much.

Not to be awkward, but what the client showed me as "see, this is nice" was the current Drupal 8 version of Seven. At least it's an example. I have no real understanding of design, so I hope these comments make some kind of general sense.

---
Milestone candidate for 1.12.2

---

THIS ISSUE:
* the size of the FORMATTING OPTIONS fieldset label. 

BEFORE:
<img width="770" alt="screen shot 2019-02-03 at 1 08 06 am" src="https://user-images.githubusercontent.com/397895/52174941-2b5ea280-2750-11e9-8f47-67af6d671b36.png">

AFTER:
<img width="632" alt="screen shot 2019-02-03 at 1 07 08 am" src="https://user-images.githubusercontent.com/397895/52174930-0a964d00-2750-11e9-9ec4-71000e590d8d.png">

RELATED:
* #3520 | on a text field with unlimited values, the "ADD ANOTHER ITEM" button is the same size, shape, and color as the field title.

CONTRIB:
* Inside a field group container like a fieldset or vertical tab, the field title isn't bolded.

---
PR by @jenlampton: https://github.com/backdrop/backdrop/pull/2498

**Comment by klonos** [comment]
Thanks for taking the time to provide feedback @bradbulger. Any chance you can provide screenshots of the things you mention please?

**Comment by olafgrabienski** [comment]
I agree! While I like Backdrop's *Seven* theme a lot, the node form isn't very consistent in size and weight of the elements. I'll try to give more detailed information soon.

**Comment by bradbulger** [comment]
here are a few examples of Backdrop
![screen shot 2019-02-01 at 4 08 13 pm](https://user-images.githubusercontent.com/1234634/52156505-61f7c880-263d-11e9-9df1-a99e0c8ad329.png)
![screen shot 2019-02-01 at 4 08 22 pm](https://user-images.githubusercontent.com/1234634/52156513-6b813080-263d-11e9-9550-f3d187db8dd6.png)
![screen shot 2019-02-01 at 4 08 28 pm](https://user-images.githubusercontent.com/1234634/52156521-720fa800-263d-11e9-8374-8080f4a52d8c.png)

**Comment by bradbulger** [comment]
similar in drupal 8 for comparison
![screen shot 2019-02-01 at 4 07 40 pm](https://user-images.githubusercontent.com/1234634/52156533-86ec3b80-263d-11e9-9b33-f161f8691c0a.png)
![screen shot 2019-02-01 at 4 07 56 pm](https://user-images.githubusercontent.com/1234634/52156537-8bb0ef80-263d-11e9-985b-2493daf59a6f.png)
![screen shot 2019-02-01 at 4 08 04 pm](https://user-images.githubusercontent.com/1234634/52156546-94092a80-263d-11e9-9853-4f52e31b1357.png)

**Comment by klonos** [comment]
> ...on a text field with unlimited values, the "ADD ANOTHER ITEM" button is the same size, shape, and color as the field title.

Not a designer or much of an artistic person, but I gave it a go using some of the colors in the palette we already use elsewhere (first button is hover, second normal):

<img width="575" alt="screen shot 2019-02-03 at 6 21 12 pm" src="https://user-images.githubusercontent.com/2423362/52174124-986f4800-27e3-11e9-9b7f-a846a9a16fd7.png">

<img width="566" alt="screen shot 2019-02-03 at 6 27 54 pm" src="https://user-images.githubusercontent.com/2423362/52174129-a91fbe00-27e3-11e9-8ea8-9a9cd5eddfd2.png">

<img width="567" alt="screen shot 2019-02-03 at 6 31 20 pm" src="https://user-images.githubusercontent.com/2423362/52174131-b341bc80-27e3-11e9-98a9-fcda64822053.png">

What do you think? Improved a bit?

**Comment by klonos** [comment]
> Inside a field group container like a fieldset or vertical tab, the field title isn't even bolded.

Are you using https://github.com/backdrop-contrib/field_group for that grouping? If so, then this should be reported in https://github.com/backdrop-contrib/field_group/issues

**Comment by klonos** [comment]
> Another example the client raised was the size of the FORMATTING OPTIONS fieldset label. The default `legend` font size is 120% - that's fine for fieldsets containing fields, but in a largely information-only box repeated for every formatted textarea on the form, it's a bit much.

By default in D7 the formatting options were not collapsed, so people would need to use https://www.drupal.org/project/hide_formats, or subtheme Seven and customise it. This was addressed in core for both Backdrop and D8. Here's the issue where this was fixed in Backdrop in #938, and then we also filed #1089 as a follow-up to improve things further. No activity there in a few years now, but I feel that this should be brought up there.

**Comment by jenlampton** [comment]
> Another example the client raised was the size of the FORMATTING OPTIONS fieldset label. The default legend font size is 120% - that's fine for fieldsets containing fields, but in a largely information-only box repeated for every formatted textarea on the form, it's a bit much.

In usability tests the drop-down had proven more distracting than a link, so we're still a step ahead of D8 in terms of UX, even if not in terms of design. But I'd be all in favor of removing the capitalization and decreasing the font size for this link, too.

**Comment by jenlampton** [comment]
Okay, PR up for review: https://github.com/backdrop/backdrop/pull/2498

**Comment by klonos** [comment]
Thanks @jenlampton! Looks really good on the PR sandbox 👍 ...one small problem I have noticed though is that when refreshing the page, the border line between the fieldset and the `cke_bottom` resizer gets lost:

<img width="1062" alt="screen shot 2019-02-03 at 8 53 58 pm" src="https://user-images.githubusercontent.com/2423362/52175330-2c4a0f80-27f6-11e9-95ef-bbcb5e82b52e.png">

...clearing caches fixes this. Refreshing the page causes it to happen again.

**Comment by klonos** [comment]
Agreed on the milestone 😄

**Comment by laryn** [comment]
What do you think about adding summary text behind the formatting options link (e.g "filtered HTML")? Or is that too much?

**Comment by jenlampton** [comment]
> I have filed a PR against your branch
Merged :)

**Comment by jenlampton** [comment]
Side question: Does anyone else object to the term `Formatting options`? I mean, there are no formatting options in here. These are filtering options. Should we change that here too?

**Comment by klonos** [comment]
@jenlampton I wouldn't mind changing it to something like `Text format` or `Text filter` if that makes better sense. What did you have in mind?

**Comment by jenlampton** [comment]
In the PR I changed it to "Filtering options"

**Comment by bradbulger** [comment]
This all looks on the right track to me fwiw. Thanks for looking at it.

**Comment by klonos** [comment]
>...this is feeling like a bike shed. Shall we start another issue about what to name things? Maybe we can also clean up the other admin pages and Admin bar link too in that issue.

Sounds good 👍 

Review-wise, this looks really good. The only thing that I could not figure out is a small animation glitch when collapsing/expanding the fieldset. Not a biggie, but would be great if we sorted that one out as well.

**Comment by laryn** [comment]
Here's an in-browser mockup (no code) using the existing ".summary" div element and default styling:

![screen shot 2019-02-04 at 9 18 09 am](https://user-images.githubusercontent.com/871421/52217383-f17bc200-285d-11e9-8798-8c9157abde36.jpg)

**Comment by jenlampton** [comment]
I like it!

**Comment by klonos** [comment]
> ...using the existing ".summary" div element

It's actually a `<span>` 😄

...I would prefer it if we maintained as much styling consistency as possible with the respective descriptions of the vertical tabs when on mobile:

<img width="633" alt="screen shot 2019-02-06 at 6 48 58 am" src="https://user-images.githubusercontent.com/2423362/52300318-a8ea1300-29db-11e9-8167-6dcf223844d8.png">

So, gray color 👍 ...although we should match the one in the vtabs. So change from `#999` to `#666`

I would also like it better if you could figure out a way to keep it aligned to the left. I realise that we cannot have it wrap under the fieldset title, because it would then defeat the purpose of this issue here (to reduce the height of that fieldset). Hmm 🤔

**Comment by klonos** [comment]
...here's a crazy idea:

<img width="573" alt="screen shot 2019-02-06 at 7 13 46 am" src="https://user-images.githubusercontent.com/2423362/52301581-d71d2200-29de-11e9-8f62-ead655d554a9.png">

So basically use the same style/concept as the "more" thingies we have in the modules listing page, which now have cute little blue arrows (#3456).

**Comment by jenlampton** [comment]
> So, gray color 👍 ...although we should match the one in the vtabs. So change from #999 to #666

I disagree with this. The point of this issue is to make this setting less distracting, so I think having it blend in with it's surroundings is the best way to do that. Changing the gay color so that it doesn't match CKEditor is going to make it stand out too much.

> basically use the same style/concept as the "more" thingies we have in the modules listing page, which now have cute little blue arrows

And get rid of the fieldset?  hm...

**Comment by klonos** [comment]
...here's the same mockup with `#999999`:

<img width="569" alt="screen shot 2019-02-06 at 7 18 32 am" src="https://user-images.githubusercontent.com/2423362/52301830-7e01be00-29df-11e9-96ed-749bd9e39e01.png">

**Comment by jenlampton** [comment]
Oh, are you talking text color and not background color? Yeah, I'm okay with this :)

**Comment by olafgrabienski** [comment]
I like the PR, it's so much better than the current capitalized link, and I agree to discuss the name of the link separately.

That said, I'd like to put a terminology question up for discussion which IIRC hasn't been considered so far: In most of my sites, for content editors, the `Formatting options` aren't *options* at all. Usually they only need the "Filtered HTML" permission, and there is no *option* in the fieldset but only *information*. Here's a screenshot how it looks (with the current PR):

<img alt="screen-backdrop-editor-without-options" src="https://user-images.githubusercontent.com/14188307/52307835-fe063480-299b-11e9-8fe9-647f4dc77991.png">

So, in my opinion, the name has two problems: its first part: *Formatting* vs. *Filter(ing)* vs. *Editor*, and its second part: *options* vs. *information*.

At the moment, I'd suggest to follow the approach of @laryn in https://github.com/backdrop/backdrop-issues/issues/3516#issuecomment-460287284 which displays the format, but to reduce the actual wording to `Editor` without any suffix:

<img alt="screen-backdrop-editor-without-options-renamed" src="https://user-images.githubusercontent.com/14188307/52308677-37d83a80-299e-11e9-94fd-5e2da6f70357.png">

With this name, the link makes also sense when there are no options:

<img alt="screen-backdrop-editor-without-options-renamed-open" src="https://user-images.githubusercontent.com/14188307/52308699-49214700-299e-11e9-8bf3-178b818228a2.png">

**Comment by laryn** [comment]
@olafgrabienski How about "Editor details"?

**Comment by olafgrabienski** [comment]
> How about "Editor details"?

Sounds also good, in my opinion!

**Comment by quicksketch** [comment]
The PR at https://github.com/backdrop/backdrop/pull/2498 is using `px`, whereas all other fonts in Seven use `em`. With the discussion continuing in this issue, I'm going to bump this to the next release. Great ideas here.

**Comment by klonos** [comment]
...as also mentioned over Gitter, I think that we could fix the style in 1.12.2 so that it is not distracting, and leave the summary/description + the renaming of the fieldset label for 1.12.3

**Comment by klonos** [comment]
@jenlampton I left a comment in the PR re sticking to only CSS changes for 1.12.2 and leaving the discussion re the fieldset summary/description, as well as the renaming of its label for 1.12.3

Also, the changes seem to have now broken things:

<img width="1072" alt="screen shot 2019-02-09 at 10 56 37 am" src="https://user-images.githubusercontent.com/2423362/52512578-a3016580-2c59-11e9-9909-1c148a9ee2ee.png">

**Comment by jenlampton** [comment]
doh. okay - will look into it.

**Comment by jenlampton** [comment]
I think your screenshot @klonos is from the broken sandboxes? Closing PR for a new one...

**Comment by klonos** [comment]
...still looks broken to me on the sandbox:

<img width="1195" alt="screen shot 2019-02-09 at 11 16 34 am" src="https://user-images.githubusercontent.com/2423362/52513028-3045b980-2c5c-11e9-9514-5493e7c431eb.png">

Your changes though do not justify this breakage :/

**Comment by jenlampton** [comment]
changing to NR until the sandboxes are back up...

**Comment by klonos** [comment]
That was fun 😄 ...RTBC ...finally!

**Comment by quicksketch** [comment]
I merged https://github.com/backdrop/backdrop/pull/2498 because I think it's an improvement, but there is still lots of room for iteration here. I'm marginally concerned that this reduced the border on the filter area to 1px instead of the 2px we use everywhere else, but that brings it inline with CKEditor. Well, let's keep making changes to improve this further. I've merged that PR into 1.x and 1.12.x. Thanks!
